pallets/runtime/src/bridge.rs

```diff
@@ -418,69 +389,102 @@ decl_module! {
+        untrusted_manual_retry: bool,
+    ) -> DispatchResult {
+        let mut tx_details = Self::bridge_tx_details(&bridge_tx.recipient, &bridge_tx.nonce);
+        // NB: This function does not care if a transaction is timelocked. Therefore, this should only be called
+        // after timelock has expired or timelock is to be bypassed by an admin.
+        ensure!(
+            tx_details.status != BridgeTxStatus::Frozen
+                && tx_details.status != BridgeTxStatus::Handled,
+            Error::<T>::ProposalAlreadyHandled
+        );
+
+        if Self::frozen() {
+            // Untruested manual retries not allowed during frozen state.
+            ensure!(!untrusted_manual_retry, Error::<T>::Frozen);
+            // Bridge module frozen. Retry this tx again later.
+            return Self::handle_bridge_tx_later(&bridge_tx, Self::timelock());
+        }
+
+        let amount = if untrusted_manual_retry {
+            // NB: The amount should be fetched from storage since the amount in `bridge_tx`
+            // may be altered in a manual retry
+            tx_details.amount
         } else {
-            <HandledTxs<T>>::insert(&bridge_tx, true);
+            bridge_tx.amount
+        };
+        if Self::issue(&bridge_tx.recipient, &amount).is_ok() {
+            tx_details.status = BridgeTxStatus::Handled;
+            tx_details.execution_block = <system::Module<T>>::block_number();
+            <BridgeTxDetails<T>>::insert(&bridge_tx.recipient, &bridge_tx.nonce, tx_details);
             Self::deposit_event(RawEvent::Bridged(bridge_tx));
+        } else if !untrusted_manual_retry {
+            // NB: If this was a manual retry, tx's automated retry schedule is not updated.
+            // Recipient missing CDD or limit reached. Retry this tx again later.
+            return Self::handle_bridge_tx_later(&bridge_tx, Self::timelock());
         }
         Ok(())
     }
 
     /// Handles a bridge transaction proposal after `timelock` blocks.
     fn handle_bridge_tx_later(
-        bridge_tx: BridgeTx<T::AccountId, T::Balance>,
+        bridge_tx: &BridgeTx<T::AccountId, T::Balance>,
         timelock: T::BlockNumber,
     ) -> DispatchResult {
+        let mut already_tried = 0;
+        let mut tx_details = Self::bridge_tx_details(&bridge_tx.recipient, &bridge_tx.nonce);
+        match tx_details.status {
+            BridgeTxStatus::Absent => {
+                tx_details.status = BridgeTxStatus::Timelocked;
+                tx_details.amount = bridge_tx.amount.clone();
+            }
+            BridgeTxStatus::Pending(x) => {
+                tx_details.status = BridgeTxStatus::Pending(x + 1);
+                already_tried = x + 1;
+            }
+            BridgeTxStatus::Timelocked => {
+                tx_details.status = BridgeTxStatus::Pending(1);
+                already_tried = 1;
+            }
+            // Already handled/frozen.
+            _ => {
```

Comment:
The `Frozen` case might better be separated from the `Handled` case. Sharing the same error message between the two can confuse people.